lib/pbench/server/api/resources/users_api.py

```diff
@@ -0,0 +1,534 @@
+            )
+            abort(403, message="Provided username is already in use.")
+
+        password = user_data.get("password")
+        if not password:
+            self.logger.warning("Missing password field")
+            abort(
+                400, message="Missing password field",
+            )
+
+        email_id = user_data.get("email")
+        if not email_id:
+            self.logger.warning("Missing email field")
+            abort(
+                400, message="Missing email field",
+            )
+        # check if provided email already exist
+        try:
+            user = User.query(email=user_data.get("email_id"))
+        except Exception:
+            self.logger.exception("Exception while querying user")
+            abort(500, message="INTERNAL ERROR")
+        if user:
+            self.logger.warning("A user tried to re-register. Email: {}", user.email)
+            abort(403, message="Provided email is already in use")
+
+        first_name = user_data.get("first_name")
+        if not first_name:
+            self.logger.warning("Missing firstName field")
+            abort(
+                400, message="Missing firstName field",
+            )
+
+        last_name = user_data.get("last_name")
+        if not last_name:
+            self.logger.warning("Missing lastName field")
+            abort(
+                400, message="Missing lastName field",
+            )
+
+        try:
+            user = User(
+                username=username,
+                password=password,
+                first_name=first_name,
+                last_name=last_name,
+                email=email_id,
+            )
+
+            # insert the user
+            user.add()
+            self.logger.info(
+                "New user registered, username: {}, email: {}", username, email_id
+            )
+
+            response_object = {
+                "status": "success",
+                "message": "Successfully registered.",
+            }
+            response = jsonify(response_object)
+            response.status_code = 201
+            return make_response(response, 201)
+        except EmailNotValidError:
+            self.logger.warning("Invalid email {}", email_id)
+            abort(400, message=f"Invalid email: {email_id}")
+        except Exception:
+            self.logger.exception("Exception while registering a user")
+            abort(500, message="INTERNAL ERROR")
+
+
+class Login(Resource):
+    """
+    Pbench API for User Login or generating an auth token
+    """
+
+    def __init__(self, config, logger, auth):
+        self.server_config = config
+        self.logger = logger
+        self.auth = auth
+        try:
+            self.token_expire_duration = self.server_config.get(
+                "pbench-server", "token_expiration_duration"
+            )
+        except NoOptionError:
+            self.token_expire_duration = None
```

Comment:
Ah now that you mention I realized we no longer are using `token_expire_duration` inside `registerUser` since we dont log the user in when they just register, I should remove that.